cluster/sbin/snappy-nodes.sh

```diff
@@ -338,12 +338,51 @@ function getNumLeadsOnHost() {
   echo $numLeadsOnHost
 }
 
+function scpConf() {
+ 
+ currentNodeIpAddr=$(ip addr | grep 'state UP' -A2 | head -n3 | tail -n1 | awk '{print $2}' | cut -f1  -d'/')
+ currentNodeHostName=$(uname -n)
+
+ if [[ "$host" != "$currentNodeIpAddr" && "$host" != "localhost" && "$host" != $currentNodeHostName ]] ; then
+  	#loop to get the all the files avaliable in Conf directory
+	for entry in "${SPARK_CONF_DIR}"/*
+	do
+	  if [ -f "$entry" ];then
+	    #${file%.*} to get the filename without the extension and ${file##*.} to get the extension alone
+	    fileName=$(basename $entry)
+	    #echo "fileName:  $fileName"
+   	    template=".template"
+	    #check the extension, interested in files those doesn't have template extension
+	    if [[ ! "$fileName" = @(*.template) ]]; then	       	
+		if ! ssh $host "test -e $entry"; then #"File does not exist."	  
+		  scp ${SPARK_CONF_DIR}/$fileName  $host:${SPARK_CONF_DIR}
+		else
+		  #file exits but before replacing with new one, create a backup file and warn user about it then replace
+      		  # except locators,leads and servers file has to replace all the time			 
+		  if [[ "$fileName" == "locators" || "$fileName" == "servers" || "$fileName" == "leads"  ]]; then
+		    scp ${SPARK_CONF_DIR}/$fileName  $host:${SPARK_CONF_DIR} 
+		  else
+	          	if [[ ! -z $(ssh $host "cat $entry" | diff - "$entry") ]] ; then
+				backupFileName=${fileName%.*}_`date +%Y_%m_%d_%H_%M_%S`.${fileName#*.}
+				echo "###### As $fileName differ from the $fileName present on $host, creating a backup file with the name $backupFileName before replacing #######"
+				ssh $host "mv ${SPARK_CONF_DIR}/$fileName ${SPARK_CONF_DIR}/$backupFileName"		    
```

Comment:
Also, as discussed, please move the backed up files inside a directory (say, "backup") in the remote conf directory.

Reply:
Modified as per review comment